<?xml version="1.0"?>
<launch>
  <param name="scan_line" value="16" />
  <param name="use_sim_time" value="true"/>
  <param name="scan_period" value="0.1" />
  <param name="vertical_angle" type="double" value="2.0" />
  <param name="max_dis" type="double" value="90.0" />
  <param name="min_dis" type="double" value="3.0" />
  <!-- <remap from="velodyne_points" to="/lslidar_point_cloud"/> -->

  <!-- Visualization -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find amcl3d)/rviz/amcl3d_floam.rviz"/>
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_velodyne" args="0 0 0 0 0 0     /base_link    /velodyne   10" />

  <node pkg="amcl3d" type="floam_odomEstimation_amcl3d" name="floam_odomEstimation_amcl3d" output="screen"/>
  <node pkg="amcl3d" type="floam_laserProcessing_amcl3d" name="floam_laserProcessing_amcl3d" output="screen"/>
  
  <include file="$(find amcl3d)/launch/amcl3d.launch">
    <arg name="map_name_path"   value="$(find amcl3d)/data/"/>
    <arg name="init_x"          value="0"/>
    <arg name="init_y"          value="0"/>
    <arg name="init_z"          value="0"/>
    <arg name="init_a"          value="0"/>
    <arg name="num_particles"   value="600"/>
    <arg name="alpha"           value="0.5"/>
    <arg name="take_off_height" value="-0.10"/>
  </include>

</launch>
